Common/Securities/IBuyingPowerModel.cs

```diff
@@ -37,6 +37,27 @@ public interface IBuyingPowerModel
         /// <param name="leverage">The new leverage</param>
         void SetLeverage(Security security, decimal leverage);
 
+        /// <summary>
+        /// Gets the margin currently allocated to the specified holding
+        /// </summary>
+        /// <param name="parameters">An object containing the security and holdings quantity/cost/value</param>
+        /// <returns>The maintenance margin required for the provided holdings quantity/cost/value</returns>
+        MaintenanceMargin GetMaintenanceMargin(MaintenanceMarginParameters parameters);
+
+        /// <summary>
+        /// The margin that must be held in order to increase the position by the provided quantity
+        /// </summary>
+        /// <param name="parameters">An object containing the security and quantity</param>
+        /// <returns>The initial margin required for the provided security and quantity</returns>
+        InitialMargin GetInitialMarginRequirement(InitialMarginParameters parameters);
+
+        /// <summary>
```

Comment:
- These 3 new methods in the interface refer to `Margin` versus other methods that refer to `BuyingPower` which is more abstract?
This question took me to review the CashPBM which does not implement any of these 3 methods and will use the underlying implementation which might not be right, since it's margin-based?

- Having `GetInitialMarginRequiredForOrder` and `GetInitialMarginRequirement` _might_ be tricky, because this last one does not take into account order fees which the first one does, same as `HasSufficientBuyingPowerForOrder`. So the result of `GetInitialMarginRequirement` might not be completely valid by it self since its missing fees?